From 2b8c513bf3b462ea8b06f0c933b4379c7ca691b7 Mon Sep 17 00:00:00 2001
From: Demon Singur <demonsingur@gmail.com>
Date: Thu, 23 Aug 2018 03:55:12 +0000
Subject: [PATCH 1/2] sepolicy: Add rules for sudaemon

This is a squash of the following commits.

commit b20a9bffcd27ff6f70cf253a6d79d286cf87c43d
Author: Tom Marshall <tdm.code@gmail.com>
Date:   Fri Feb 16 23:13:54 2018 +0100

    sepolicy: Allow recovery to write to rootfs

     * Needed for adb_keys, fstab

    Change-Id: I72c8bc56d4c8383d7921e94edd07fea08b25eb4a

commit b5f84cab1166eaafbbd3e993aecce47cf4d87f75
Author: jrior001 <jriordan001@gmail.com>
Date:   Mon Jan 8 17:52:16 2018 -0500

    sepolicy: add sudaemon to ignore list

    Change-Id: I02c1699d13f88805a5656a45a1b47743f3f07d04

commit 85e5adcd9d919aeeae496324baa6684a5238286e
Author: jrior001 <jriordan001@gmail.com>
Date:   Mon Jan 8 01:15:22 2018 -0500

    sepolicy: update policies for sudaemon on O

    Change-Id: I552e6b0d5358d4bce20164e67721e319abcd16ce

commit 1975e0c1927360beec6ee20b082aade04967f706
Author: Steve Kondik <steve@cyngn.com>
Date:   Fri Aug 26 04:29:17 2016 -0700

    sepolicy: Allow su by apps on userdebug_or_eng

     * Our framework deals with access control.

    Change-Id: Idc737c545c358752b5c5aea468fa50525e7bdd42

commit 06dcf5b6297d42a13dc5cdb9c5edb2724587edaf
Author: Steve Kondik <steve@cyngn.com>
Date:   Fri Oct 16 20:05:10 2015 -0700

    sepolicy: We need to declare before referencing

Change-Id: I91cd6166e2f312c9bc6c9079d439cd2579aa3956
Signed-off-by: Satish Patel <tosatishpatel@gmail.com>
---
 public/domain.te | 11 ++++++-----
 public/su.te     |  2 +-
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/public/domain.te b/public/domain.te
index 987bb9f..db59afc 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -488,7 +488,7 @@ neverallow * exec_type:dir_file_class_set mounton;
 neverallow { domain -init } { system_file_type vendor_file_type }:dir_file_class_set mounton;
 
 # Nothing should be writing to files in the rootfs.
-neverallow * rootfs:file { create write setattr relabelto append unlink link rename };
+neverallow { domain userdebug_or_eng(`-recovery') } rootfs:file { create write setattr relabelto append unlink link rename };
 
 # Restrict context mounts to specific types marked with
 # the contextmount_type attribute.
@@ -796,6 +796,8 @@ full_treble_only(`
     -init
     -ueventd
     -socket_between_core_and_vendor_violators
+    -sudaemon
+    -system_app
   } {
     file_type
     dev_type
@@ -1108,7 +1110,7 @@ neverallow * { file_type fs_type dev_type }:{ lnk_file fifo_file sock_file } mou
 # Nobody should be able to execute su on user builds.
 # On userdebug/eng builds, only dumpstate, shell, and
 # su itself execute su.
-neverallow { domain userdebug_or_eng(`-dumpstate -shell -su') } su_exec:file no_x_file_perms;
+neverallow { domain userdebug_or_eng(`-dumpstate -shell -su -sudaemon -domain -init') } su_exec:file no_x_file_perms;
 
 # Do not allow the introduction of new execmod rules. Text relocations
 # and modification of executable pages are unsafe.
@@ -1192,9 +1194,8 @@ neverallow {
   -zygote
 } shell:process { transition dyntransition };
 
-# Only domains spawned from zygote, runas and simpleperf_app_runner may have the appdomain
-# attribute.
-neverallow { domain -simpleperf_app_runner -runas -app_zygote -webview_zygote -zygote } {
+# Only domains spawned from zygote and runas may have the appdomain attribute.
+neverallow { domain -runas -webview_zygote -zygote } {
   appdomain -shell userdebug_or_eng(`-su')
 }:process { transition dyntransition };
 
diff --git a/public/su.te b/public/su.te
index a2f435e..a6b7b4a 100644
--- a/public/su.te
+++ b/public/su.te
@@ -3,7 +3,7 @@
 type su, domain;
 
 # File types must be defined for file_contexts.
-type su_exec, system_file_type, exec_type, file_type;
+type su_exec, exec_type, file_type;
 
 userdebug_or_eng(`
   # Domain used for su processes, as well as for adbd and adb shell
-- 
2.17.1

